src/Symfony/Component/Intl/Data/Generator/CurrencyDataGenerator.php

```diff
@@ -48,6 +48,54 @@ class CurrencyDataGenerator extends AbstractDataGenerator
      */
     private array $currencyCodes = [];
 
+    /**
+     * Decodes ICU "date pair" into a DateTimeImmutable (UTC).
+     *
+     * ICU stores UDate = milliseconds since 1970-01-01T00:00:00Z in a signed 64-bit.
+     *
+     * @param array{0: int, 1: int} $pair
+     *
+     * @return non-empty-string
+     */
+    private static function icuPairToDate(array $pair): string
+    {
+        [$highBits32, $lowBits32] = $pair;
+
+        // Recompose a 64-bit unsigned integer from two 32-bit chunks.
+        $unsigned64 = ((($highBits32 & 0xFFFFFFFF) << 32) | ($lowBits32 & 0xFFFFFFFF));
+
+        // Convert to signed 64-bit (two's complement) if sign bit is set.
+        if ($unsigned64 >= (1 << 63)) {
+            $unsigned64 -= (1 << 64);
+        }
+
+        // Split into seconds and milliseconds.
+        $seconds = intdiv($unsigned64, 1000);
+        $millisecondsRemainder = $unsigned64 - $seconds * 1000;
+
+        // Normalize negative millisecond remainders (e.g., for pre-1970 values)
+        if (0 > $millisecondsRemainder) {
+            $millisecondsRemainder += 1000;
+
+            --$seconds;
+        }
+
+        // Build "U.u" string (seconds.microseconds). Pad ms to 3 digits, then add 000 to get microseconds.
+        $epochWithMicros = \sprintf('%d.%03d000', $seconds, $millisecondsRemainder);
+
+        $datetime = \DateTimeImmutable::createFromFormat(
+            'U.u',
```

Comment:
let's simplify this and remove the part about $millisecondsRemainder
in the end we have only day precision